business/istio_config.go

```diff
@@ -285,18 +217,40 @@ func (in *IstioConfigService) GetIstioConfigList(criteria IstioConfigCriteria) (
 	go func(errChan chan error) {
 		defer wg.Done()
 		if criteria.IncludeTemplates {
-			if tt, ttErr := in.k8s.GetTemplates(criteria.Namespace, ""); ttErr == nil {
+			if tt, ttErr := in.k8s.GetIstioObjects(criteria.Namespace, kubernetes.Templates, ""); ttErr == nil {
 				istioConfigList.Templates = models.CastIstioTemplatesCollection(tt)
 			} else {
 				errChan <- ttErr
 			}
 		}
 	}(errChan)
 
+	go func(errChan chan error) {
+		defer wg.Done()
+		if criteria.IncludeHandlers {
+			if hh, hhErr := in.k8s.GetIstioObjects(criteria.Namespace, kubernetes.Handlers, ""); hhErr == nil {
+				istioConfigList.Handlers = models.CastIstioHandlersCollection(hh)
+			} else {
+				errChan <- hhErr
+			}
+		}
+	}(errChan)
+
+	go func(errChan chan error) {
+		defer wg.Done()
+		if criteria.IncludeInstances {
+			if ii, iiErr := in.k8s.GetIstioObjects(criteria.Namespace, kubernetes.Instances, ""); iiErr == nil {
+				istioConfigList.Instances = models.CastIstioInstancesCollection(ii)
```

Comment:
I can see sometimes we use `CastXXX` and other times `xxx.Parse`, I believe this is two different ways to do the same thing? If so, that's something we could refactor too? (here or another PR)

Personally I prefer the `CastXXX` way as it handles object initialization.

Reply:
Yes, there is some technical debt in that area, due mixed objects.
At it's not really urgent, I left that untouched until we have some strong changes in that area and we can apply it.